pkg/filter/filter_test.go

```diff
@@ -0,0 +1,349 @@
+package filter_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	"github.com/openshift-psap/special-resource-operator/api/v1beta1"
+	"github.com/openshift-psap/special-resource-operator/pkg/filter"
+	buildv1 "github.com/openshift/api/build/v1"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+func TestFilter(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Filter Suite")
+}
+
+var _ = Describe("Filter", func() {
+	Context("SetLabel", func() {
+		objs := []runtime.Object{
+			&v1.DaemonSet{
+				TypeMeta: metav1.TypeMeta{Kind: "DaemonSet"},
+			},
+			&v1.Deployment{
+				TypeMeta: metav1.TypeMeta{Kind: "Deployment"},
+			},
+			&v1.StatefulSet{
+				TypeMeta: metav1.TypeMeta{Kind: "StatefulSet"},
+			},
+		}
+
+		entries := make([]TableEntry, 0, len(objs))
+
+		for _, o := range objs {
+			entries = append(entries, Entry(o.GetObjectKind().GroupVersionKind().Kind, o))
```

Comment:
i think that adding a string parameter that describe the test ("DaemonSet", "Deployment", or "StatefulSet") is needed